tests/fixtures/dummy_project/pyproject.toml

```diff
@@ -18,11 +18,30 @@ cmd = "echo POE_ROOT:$POE_ROOT ${BEST_PASSWORD}, task_args:"
 help = "It says what you say"
 env = { BEST_PASSWORD = "Password1" }
 
+[tool.poe.tasks.greet-rekeyed]
+script = "dummy_package:args(greeting, user,optional=opt, upper=upper, fvar, ivar)"
+  [tool.poe.tasks.greet-rekeyed.args.greeting]
+    type = 'string'
+  [tool.poe.tasks.greet-rekeyed.args.opt]
+  [tool.poe.tasks.greet-rekeyed.args.user]
+    type = 'string'
+  [tool.poe.tasks.greet-rekeyed.args.upper]
+    type = 'boolean'
+  [tool.poe.tasks.greet-rekeyed.args.ivar]
+    type = 'integer'
+  [tool.poe.tasks.greet-rekeyed.args.fvar]
+    type = 'float'
+    help ='A float'
+
+
 [tool.poe.tasks]
 show_env = "env"
 greet = { script = "dummy_package:main" }
 greet-shouty = { script = "dummy_package:main(upper=True)" }
-count = { shell = "echo 1 && echo 2 && echo $(python -c 'print(1 + 2)')" }
+greet-keyed = { script ="dummy_package:args(greeting, user, optional, upper, default=default_value)", args = ["greeting", "user", "optional", "upper"]}
```

Comment:
What's default_value here?

Reply:
> What's default_value here?

I think that's an interesting question that deserves a bit more detail because this is where behavior becomes a bit convoluted IMO.

`default=default_value`

a TLDR: This will be parsed as default=Str(default_value) and passed as such to the kwargs

---------------
A bit longer
When we pass this to script.py@102 we pass `call_params` and `named_params` the latter will have only the (4) arguments named in the square brackets of the TOML file the former will have all (5) arguments defined in the parentheses. So any value defined in those default values will by default be defined as a string object. The default=default_value will not be present in `argv` but it will be present in the `call_params` we actually pass off to the final python function script.py@54.

In summary for this bit the `argv` will only contain arguments defined the square brackets. This might allow a user to have some pre-defined behavior for a function with only some optional components. The downside of this is that any arguments not defined specifically in the TOML will be strings -- Thus any attempt at a Boolean flag will fail if a user anticipates that `flag=false` evaluates to `False`

Reply:
> What's default_value here?

In this case default_value is "default_value" to poe. The default is sent to the `call_params` rather than `argv`

```        
        cmd = (
            "python",  # TODO: pre-locate python from the target env?
            "-c",
            "import sys; "
            "from importlib import import_module; "
            f"sys.argv = {argv!r}; sys.path.append('src');"
            f"import_module('{target_module}').{target_callable}({call_params or ''})",
        )
```



A simple example of why I could see my self using this pattern of behavior 
```
morning = { script = "dummy_package:main(greeting=Good Morning, name)" , args = ["name"] }
evening = { script = "dummy_package:main(greeting=Good Evening, name)" , args = ["name"] }
```